packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,83 @@
-use std::path::PathBuf;
-
 use memory::shared_memory::MemoryId;
 
 use crate::{
-    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result},
+    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result, Error},
     proto::ExperimentId,
 };
 
+// TODO: move this to top-level
+#[derive(PartialEq, Eq, Clone)]
+pub enum EngineExitStatus {
+    Success,
+    Error,
+}
+
 // TODO: move this to top-level
 /// Shared memory cleanup in the process hard crash case.
 /// Not required for pod instances.
-pub fn cleanup_experiment(experiment_id: &ExperimentId) -> Result<()> {
+pub fn cleanup_experiment(
+    experiment_id: &ExperimentId,
+    exit_status: EngineExitStatus,
+) -> Result<()> {
     MemoryId::clean_up(experiment_id)?;
 
-    // TODO: We don't want to be deleting the parts files by default. We should figure out what to
-    //   do with this.
-    // remove_experiment_parts(experiment_id).or_else(|e| {
-    //     let err_string = e.to_string();
-    //     if err_string.contains("kind: NotFound") {
-    //         Ok(())
-    //     } else {
-    //         Err(Error::from(format!(
-    //             "Error while trying to remove experiment parts folder: {err_string}"
-    //         )))
-    //     }
-    // })?;
+    // Cleanup python socket files in case the engine didn't
+    let frompy_files = glob::glob(&format!("{experiment_id}-frompy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+    let topy_files = glob::glob(&format!("{experiment_id}-topy*"))
+        .map_err(|err| Error::Unique(format!("cleanup glob error: {err}")))?;
+
+    frompy_files
+        .chain(topy_files)
+        .filter_map(std::result::Result::ok)
+        .for_each(|path| match std::fs::remove_file(&path) {
+            Ok(_) => {
+                match exit_status {
+                    EngineExitStatus::Success => tracing::event!(
+                        tracing::Level::ERROR,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                    EngineExitStatus::Error => tracing::event!(
+                        tracing::Level::TRACE,
```

Comment:
Let's make this a warning
```suggestion
                    EngineExitStatus::Error => tracing::warn!(
```